packages/react-app/src/contexts/Web3Context.jsx

```diff
@@ -50,6 +56,15 @@ export const Web3Provider = ({ children }) => {
   const { providerChainId, ethersProvider } = web3State;
   const [account, setAccount] = useState();
   const [loading, setLoading] = useState(true);
+  const [isChainIdInjected, setIsChainIdInjected] = useState({
```

Comment:
We can just use chainId in Web3Context

Reply:
Again, this seems to contradict with the implementation requirement, as (if I have understood it right) providerChainId is something which is over metamask at the current instance, and not referring thebquery params right?

Reply:
Sorry. I guess I didn't make myself clear. I am suggesting that we do the chainId query param extraction in Web3Context and token in BridgeContext

Reply:
Got it..